src/app/shared/user/user.service.ts

```diff
@@ -74,14 +80,28 @@ export class UserService {
     }
   }
 
-  getSingleUser(): void {
+  getSingleUser(redirectAfterSuccess: boolean, url?: string): void {
     this.setupConfigurationToken();
     if (this.configuration.apiKeys['Authorization']) {
       this.usersService.getUser().subscribe(
         (user: User) => {
           this.updateUser(user);
           if (user) {
             this.tokenService.get(user.id);
+            if (redirectAfterSuccess) {
+              switch (url) {
+                case '/orcid':
+                  this.orcidAccountIsLinked$ = this.tokenQuery.hasOrcidToken$;
```

Comment:
Rusty on Akita, not sure if these make sense:

1. Does this need to be a class property? Can't `this.orcidAccountIsLinked$` be a local variable?
2. Do you need to unsubscribe?
3. Could you just get the value instead of using a subscription? Or do you expect the subscription to become true at some point after this?

Reply:
1. no, yes
2. yes
3. should be yes